### PR TITLE
Additional_nodes DTWF/FIXED_PEDIGREE

### PR DESCRIPTION
**Pull request by GertjanBisschop**
`algorithms.py` outline for tracking `additional_nodes` for `DTWF` and `FIXED_PEDIGREE` models.

### CODE REVIEW

File: algorithms.py [hunk 1]
```diff
@@ -771,6 +771,12 @@ def flush_edges(self):
             self.tables.edges.add_row(left, right, parent, child)
             self.edge_buffer = []
 
+    def update_node_flag(self, node_id, flag):
+        self.flush_edges()
```
Comment (jeromekelleher) [inline algorithms.py:775]: Do we need this flush_edges? Should probably be done in the other function we just added?

Comment (GertjanBisschop) [reply algorithms.py:775]: Adding `flush_edges()` to `store_additional_nodes_edges()` would flush the edges twice in some cases, as it is also called by `store_node()`.

Comment (jeromekelleher) [reply algorithms.py:775]: I see - should we call flush_edges to the non-store-node path in store_additional_nodes_edges then?

Comment (jeromekelleher) [reply algorithms.py:775]: It's a surprising side-effect to have here in ``update_node_flag``

**Review by jeromekelleher — APPROVED** [review]
LGTM!

**Pull request merged**